File: C++/bug_reproducer_assistant/annotator_test/MyFunctions.cpp
//Boost
#include <boost/lexical_cast.hpp>
//annotator_test
#include "MyFunctions.h"

using namespace std;

//Empty print for not disturbing the unit tests output
void myPrint(const string& str)
{}

int add(int i,  int j)
{
	return i + j;
}
	
int subtract(int i, int j)
{
	return i - j;
}
	
std::string mySubstring(const string& str, size_t length)
{
	return str.substr(0, length);
}

void processVector( const vector< int >& aList )
{
	myPrint("Entered processVector");
	for ( vector< int >::const_iterator it = aList.begin(); it != aList.end(); ++it )
		myPrint ("elem: " + boost::lexical_cast< string >(*it) + "\n");
}

void processIntMap( const map<int,int>& aMap )
{
	myPrint("Entered processIntMap");
	for ( map< int, int >::const_iterator it = aMap.begin(); it != aMap.end(); ++it )
	{
		myPrint ("key: " + boost::lexical_cast< string >(it->first) + "\n");
		myPrint ("value: " + boost::lexical_cast< string >(it->second) + "\n");
	}
}

void innerFunction()
{
}

void outerFunction()
{
	innerFunction();
}

void noParamsFunction()
{
}

//MyClass
void MyClass::f1()
{
	myPrint( "No params" );
}

void MyClass::f2(int i)
{
	myPrint( "i:" + boost::lexical_cast< string > (i) );
}

void MyClass::f3(const std::vector< int > aVector)
{
	int x = aVector[0];
	myPrint( "x:" + boost::lexical_cast< string >(x) );
}
void MyClass::f4(std::map< int, int > aMap, MyClass * anObj)
{
	int x = aMap[1];
	int y = aMap[2];
	myPrint("x: " + boost::lexical_cast< string >(x)+ ", y:" + boost::lexical_cast< string >(y) );
	if ( anObj == NULL )
		myPrint("anObj is NULL");
	else
		myPrint("anObj is not NULL");
}

//ClassWithConstructor
ClassWithConstructor::ClassWithConstructor(int x, int y):
	x_(x),
	y_(y)
{
	myPrint("No params");
}

int ClassWithConstructor::getX()
{
	return x_;
}
void ClassWithConstructor::setX(int x)
{
	x_ = x;
}

int ClassWithConstructor::getY()
{
	return y_;
}

void ClassWithConstructor::setY(int y)
{
	y_ = y;
}

// ClassWithStaticMethods
void ClassWithStaticMethods::static0()
{
	myPrint("static0");
}

void ClassWithStaticMethods::static1( int x )
{
	myPrint( "x:" + boost::lexical_cast< string >(x) );
}

File: Python/tests/source_code_parser_test_files/MyFunctions.py
#Empty print for not disturbing the unit tests output
def myPrint( str ):
    pass

def add( i,  j):
    return i + j
    
def subtract( i,  j):
    return i - j

def innerFunction():
    pass

def outerFunction():
    innerFunction()

def noParamsFunction():
    return MyClass()

def mySubstring( str,  length ):
    return str[:length]

def processList( aList ):
    myPrint( "Entered processList" )
    for elem in aList:
        myPrint ("elem: " + repr(elem) + "\n")

def processDict( aDict ):
    myPrint( "Entered processDict" )
    for elem in aDict.items():
        key,  value = elem
        myPrint( "key: " + repr(key) + "\n" )
        myPrint( "value: " + repr(value) + "\n" )
        
class MyClass:
    def f1(self):
        myPrint( "No params" )
        return self
    def f2(self, i):
        myPrint( "i:" + repr(i) )
    def f3(self, aList):
        x = aList[0]
        myPrint( "x:" + repr(x) )
    def f4(self, aDict, anObj):
        x = aDict['x']
        y = aDict['y']
        myPrint("x: " + repr(x) + ", y:" + repr(y) )
        myPrint("anObj: "  + repr(anObj))
    def f5(self, obj1, obj2):
        myPrint("obj1: " + str(obj1) + ", obj2: " + str(obj2)) 

class ClassWithDummyParameters:
    def f1(self, d):
        myPrint( "Dummy parameter" )
        
class NonAnnotatedClass:
    pass

class ClassWithConstructor:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        myPrint( "No params" )
    def getX(self):
        return self.x
    def setX(self,x):
        self.x = x 
    def getY(self):
        return self.y
    def setY(self, y):
        self.y = y
        
class ClassWithStaticAndClassMethods:
    @staticmethod
    def static0():
        myPrint( "static0")
    @staticmethod
    def static1( x ):
        myPrint( "x: " + repr(x) )
    @classmethod
    def classMethod0(cls):
        myPrint( "Class is " + repr(cls) )
    @classmethod
    def classMethod1(cls,x):
        myPrint( "Class is " + repr(cls) + "x = " + str(x) )